### ISSUE TITLE
[a11y] Footer heading level and landmark issues

### ISSUE THREAD

**Issue by alexgibson** [mozilla/bedrock#15004]
## Description

There are two a11y issues on footer component related to heading levels and landmark roles.

### Steps to reproduce

1. Open https://www.mozilla.org/en-US/
2. Scroll down to the footer

### Expected result

Heading levels: these are currently `<h5>`, but should probably be an `<h2>` seeing as it follows the HTML5 outline algorithm?

Landmark roles: the two `<nav>` elements within the footer are missing a unique role/label/title.

### Actual result

![image](https://github.com/user-attachments/assets/84243505-4090-43ad-8296-2d84a3bfc256)

### Environment

N/A

**Comment by alexgibson** [comment]
Fixed in https://github.com/mozilla/bedrock/pull/15186
